audacity/audacity
# 5794 tempo detection speed improvement

Resolves: #5794 

This PR intends to speed up the tempo detection **without modifying its quality**. Guarantee that quality is not modified is given by the benchmarking, re-run after each optimization and checking that none of the dataset's 140 files was evaluated differently.

Processing time measurement was added to the benchmarking. On a Windows laptop, the algorithm is about 4 times faster after these optimizations, and on a Macbook, about 3 times. Commit-by-commit improvement is detailed in the commit messages.




Recommended:

QA:
That the quality did not change noticeably was verified with the benchmarking tests. Please nevertheless smoke-test the build, importing a few rhythmic and non-rhythmic files, preferably that you've already used before. If you have long files with high sample rate that have brought the progress bar in focus before this change, please try them again now: the speed should be noticeably faster.
